pyat/at/tracking/particles.py

```diff
@@ -125,36 +133,51 @@ def sigma_matrix(ring=None, twiss_in=None, emitx=None,
         assert espread is not None, 'espread must be defined'
 
     if ring:
+        cavPassFlag = numpy.any(
+                        numpy.array(
+                          [i.PassMethod == 'CavityPass' for i in ring]))
+        radPassFlag = numpy.any(
+                        numpy.array(
+                          ['Rad' in i.PassMethod for i in ring]))
+
+        if cavPassFlag and not radPassFlag and not flag:
+            raise AtError('Cannot compute 6D sigma matrix without '
```

Comment:
Maybe you could mention in the comment what needs to be done: Cannot compute sigma matrix: please turn on radiation or provide input emittances

Reply:
Updated